crates/libcgroups/src/v2/devices/bpf.rs

```diff
@@ -24,26 +24,32 @@ pub mod prog {
     // TODO: consider use of #[mockall_double]
     #[cfg(test)]
     use crate::v2::devices::mocks::mock_libbpf_sys::{
-        bpf_load_program, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
+        bpf_prog_load,bpf_prog_load_opts, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
     };
     #[cfg(not(test))]
     use libbpf_sys::{
-        bpf_load_program, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
+        bpf_prog_load,bpf_prog_load_opts, bpf_prog_attach, bpf_prog_detach2, bpf_prog_get_fd_by_id, bpf_prog_query,
     };
 
     pub fn load(license: &str, insns: &[u8]) -> Result<RawFd> {
         let insns_cnt = insns.len() / std::mem::size_of::<bpf_insn>();
         let insns = insns as *const _ as *const bpf_insn;
+        let opts = {
+            let mut opts = bpf_prog_load_opts::default();
+            opts.kern_version = 0;
+            opts.log_buf = ptr::null_mut::<i8>();
+            opts.log_size = 0;
+            opts
+        };
         #[allow(unused_unsafe)]
         let prog_fd = unsafe {
-            bpf_load_program(
+            bpf_prog_load(
                 BPF_PROG_TYPE_CGROUP_DEVICE,
+                ptr::null::<i8>(),
+                license as *const _ as *const i8,
                 insns,
                 insns_cnt as u64,
-                license as *const _ as *const i8,
-                0,
-                ptr::null_mut::<i8>(),
-                0,
+                &opts
```

Comment:
Logic behind this change :
- The second param is program name as string *. Now originally we used bpf_load_prog, where we can see in its [source](https://github.com/libbpf/libbpf/commit/9476dce6fe905a6bf1d4c483f7b2b8575c4ffb2d#diff-4b440c545696b4c1c729c0ebc776a5c478ebf893f01d4891c2cd7fd1694431baL430) it calls [bpf_load_program_xattr2](bpf_load_program_xattr2), in which the bpf_prog_load (named different before major release) was [called here](https://github.com/libbpf/libbpf/commit/9476dce6fe905a6bf1d4c483f7b2b8575c4ffb2d#diff-4b440c545696b4c1c729c0ebc776a5c478ebf893f01d4891c2cd7fd1694431baL426). The bpf_program_load would internally set the opts, and set the name as [null here](https://github.com/libbpf/libbpf/commit/9476dce6fe905a6bf1d4c483f7b2b8575c4ffb2d#diff-4b440c545696b4c1c729c0ebc776a5c478ebf893f01d4891c2cd7fd1694431baL440). Thus we should be able to safely do that without breaking original behavior (theoretically).
- After that the license param was moved up in the list, and the last three are set in the opts above.

Reply:
! Not Outdated, marked so because of the forced fmt push